src/services/collaborativesearch.service.ts

```diff
@@ -829,6 +878,168 @@ export class CollaborativesearchService {
         return result;
     }
 
+
+    private computeResolveWithAbort(projection: [projType.aggregate, Array<Aggregation>]
```

Comment:
If it is a repeated type, you may also use a type alias (I don't know if it is more readable, just an idea)

Reply:
it's been refactored, so that the code is not duplicated